MP - 67, MP - 66 - adding tables, boards in the administration panel

kacpergadula opened the pull request:
MP - 67 Creating a single table on the board
MP - 66 Creating a board for adding tables in the administration panel

kacpergadula committed:
feat: addition of a table board view

kacpergadula committed:
feat: Implementation React DnD

Implementation React DnD : useDrag, useDrop, add child to board

kacpergadula committed:
feat: tables

tables

kacpergadula committed:
perf: :zap: changing the logic of moving tables

change of table moving logic (free moving on the board, not on divs)

kacpergadula committed:
perf: :zap: add new Table on Board

add Fuction to add new Table on Board

kacpergadula committed:
feat: :zap: preparation for saving tables

kacpergadula committed:
perf: :zap: Merge development into feature/MP-50

Merge remote-tracking branch 'origin/development' into feature/MP-50

kacpergadula committed:
perf: :zap: table size

changing the size of the table

mfurmanit commented on src/components/PlanRoom/Board.tsx:
```diff
@@ -0,0 +1,20 @@
+    h="400px"
+    w="100%"
+    border="1px solid black"
+    bgImage="url('https://firebasestorage.googleapis.com/v0/b/mfg-pos.appspot.com/o/Board%2FbackGroundBoard.png?alt=media&token=76b04e32-6b94-4b6f-8dce-7cf512fe20a0')"
```
Generally in my Angular projects I tend to store static images / icons in separate assets folder in source code.
I don't know if here using `url` from our Firebase Storage is more appropriate.
@dominikmlynarczyk what do you think?

kacpergadula replied:
This will be changed in the next commit

mfurmanit commented on src/components/PlanRoom/Board.tsx:
```diff
@@ -0,0 +1,20 @@
+const Board = () => (
+  <Box
+    id="board"
+    h="400px"
```
In my opinion we should avoid using static `px` definitions. 
Views should be responsive based on browser width and height.
Consider using modifiers like `rem` or/and `em`.

kacpergadula replied:
I think so too, but for calculations later, the static determination of the size gives the possibility of defining the framework. Already such a problem with responsiveness to width arises. Then, it is difficult to give a frame to which the conditions will apply

![image](https://user-images.githubusercontent.com/43721352/120540291-cd5a2c80-c3e8-11eb-92ae-1a87ae1e2b01.png)

Example that size looks good on smaller devices. I would leave it at that

mfurmanit commented on src/components/PlanRoom/Board.tsx:
```diff
@@ -0,0 +1,20 @@
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import Items from './RoomTables/Item';
+
+const Board = () => (
```
It should be named `index.tsx` (and stored in separate folder named `Board`) as @dominikmlynarczyk already pointed out in our previous conversations.

mfurmanit replied:
Please apply it in all appropriate places so I don't have to write the same comment everywhere.

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+
+const Items = () => {
+  const [tables, setTables] = useState<{
+    [key: string]: { top: number; left: number; text: string; width: number; height: number; borderRadius: number };
```
You should create separate `type` definition and store it in proper file outside of this component.

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+let changeValueTableBorderRadius: Function;
+let newTable: Function;
+let save: Function;
+const backgroundColor: string = '#211F1F';
```
We should customize colors used in the app by using [Chakra Theme](https://chakra-ui.com/docs/theming/customize-theme#customizing-theme-tokens).
Please refer to the documentation and extend theme with concrete color / colors with proper naming convention.

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+  );
+
+  save = () => {
+    console.log(tables);
```
That should be deleted.

kacpergadula replied:
it is temporary

mfurmanit commented on src/views/Settings/Tables.tsx:
```diff
@@ -1,5 +1,23 @@
+const Tables = () => (
+  <Box w="100%" minWidth="30em" h="36em">
+    <Flex w="100%" p="1em">
+      <VStack w="100%" spacing={3} align="stretch">
```
We should use consistent declarations using quotes instead of braces, so you should change spacing from `{3}` to `"3"`.

kacpergadula replied:
you're right

mfurmanit commented on src/components/PlanRoom/SaveRoomTables/SaveBoard.tsx:
```diff
@@ -0,0 +1,7 @@
+const SaveBoard = (tables: {
+  [key: string]: { top: number; left: number; text: string; width: number; height: number; borderRadius: number };
+}) => {
+  console.log(tables);
```
I don't see any point in showing / implementing this component.
It does nothing except logging information to console, which we are not interested in.

kacpergadula replied:
will be deleted

mfurmanit commented on src/styled/PlanRoom/breakpointsButton.tsx:
```diff
@@ -0,0 +1,12 @@
+import { createBreakpoints } from '@chakra-ui/theme-tools';
+
+const breakpoints = createBreakpoints({
```
There is no point in defining breakpoints, where each one has the same value.

kacpergadula replied:
The file will be deleted and the parameter next to the button will be applied

mfurmanit commented on src/components/PlanRoom/SaveBoard.tsx:
```diff
@@ -0,0 +1,43 @@
+      <Box minWidth="10%">
+        <Button
+          type="submit"
+          bg="#33D268"
```
We should customize colors used in the app by using [Chakra Theme](https://chakra-ui.com/docs/theming/customize-theme#customizing-theme-tokens).
Please refer to the documentation and extend theme with concrete color / colors with proper naming convention.

Please apply it in all appropriate places so I don't have to write the same comment everywhere.

kacpergadula replied:
will be changed

mfurmanit commented on src/components/PlanRoom/SaveBoard.tsx:
```diff
@@ -0,0 +1,43 @@
+import breakpointsFontSize from 'styled/PlanRoom/breakpointsFontSize';
+import { NewTable, SaveTables } from './RoomTables/Item';
+
+const saveBoard = () => (
```
Components should start with big letter.

kacpergadula replied:
will be changed

mfurmanit commented on src/components/PlanRoom/RoomTables/Types.tsx:
```diff
@@ -0,0 +1,22 @@
+import { ReactNode } from 'react';
```
Type definitions should be stored probably in the folder called `model` as it is right now.

kacpergadula replied:
despite the transfer there is a need for this line of code to be in the file

mfurmanit commented on src/components/PlanRoom/RoomTables/Types.tsx:
```diff
@@ -0,0 +1,22 @@
+import { ReactNode } from 'react';
+
+export type TypeItem = {
```
`TableType` and `TypeItem` have many common properties. 
You should extract them to separate type and use [intersection](https://levelup.gitconnected.com/typescript-advanced-types-union-and-intersection-types-9283046d7859).

kacpergadula replied:
these types will be more integrated

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+
+let numberTable: number = 1;
+
+export const NewTable = () => {
```
Functions should start with lowercase letter.

mfurmanit replied:
Please apply it in all appropriate places so I don't have to write the same comment everywhere.

kacpergadula replied:
Zastosowałem do najnowszej wersji sugestie Dominika : PascalCase

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+let changeValueTable: Function;
+let changeValueTableBorderRadius: Function;
+let newTable: Function;
+let save: Function;
```
You should not declare variables in "global scope" of this file.
Use `state` or another (proper) hooks to do that.

kacpergadula replied:
I will leave it at that for now

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+
+  save = () => {
+    console.log(tables);
+    return tables;
```
This function is pointless if it only returns data.

kacpergadula replied:
this will be changed in further tasks

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+let save: Function;
+const backgroundColor: string = '#211F1F';
+
+const Items = () => {
```
Functions / logic contained in this file should be divided and extracted (it's to big) to proper places and couple other files.

kacpergadula replied:
True, but unfortunately I was not able to export from inside the TableList functions, e.g. AddNewTable

mfurmanit commented on src/components/PlanRoom/RoomTables/Item.tsx:
```diff
@@ -0,0 +1,184 @@
+    0: { top: 0, left: 0, text: '0', width: 100, height: 50, borderRadius: 0 },
+  });
+
+  const moveTable = useCallback(
```
You can merge those three functions one below the other to one by passing optional parameters or in some other way.
You do the same thing three times except for passing different parameters.

kacpergadula replied:
Right, but I'll leave it at that for now and move on

mfurmanit reviewed (changes requested):
Thanks a lot for the changes you have made. 🔥 💪 
I looked through most of the stuff, but I didn't cover everything yet.
Please refer to the comments and correct individual errors.

kacpergadula committed:
refactor: :art: changes after the review ( more to come leter )

renaming functions, implementing Chakra Theme, defining Tables Type externally, changing file structure, adding local board background graphic

dominikmlynarczyk commented on .eslintrc:
```diff
@@ -62,12 +62,12 @@
     "react/react-in-jsx-scope": "off",
-    "import/no-cycle": "off"
+    "import/no-cycle": "off",
+    "react/require-default-props": "off"
```
Try not to add these types of changes to commits, as you only rearranged eslint rules.

kacpergadula replied:
oki

dominikmlynarczyk commented on package.json:
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@chakra-ui/icons": "^1.0.6",
+    "@chakra-ui/layout": "^1.4.5",
```
What is the point of setting that as a dependency? I believe we can use `@chakra-ui/react` without needing to add cycled dependency.

kacpergadula replied:
Sorry, I forgot to uninstall

mfurmanit commented on src/components/molecules/Supplier/AddSupplierData/index.tsx:
```diff
@@ -11,10 +11,10 @@ import {
-import breakpoints from 'styled/Supplier/AppSupplierData/breakpoints';
-import breakpointsFontSize from 'styled/Supplier/AppSupplierData/breakpointsFontSize';
-import breakpointsMargin from 'styled/Supplier/AppSupplierData/breakpointsMargin';
+import AddSupplierButtons from '../AddSupplierButtons/index';
```
@dominikmlynarczyk said that we should be using absolute paths (just from `/src` directory).
Please apply it in all appropriate places so I don't have to write the same comment everywhere.

kacpergadula replied:
It will be corrected

mfurmanit commented on src/components/molecules/Supplier/AddSupplierData/index.tsx:
```diff
@@ -11,10 +11,10 @@ import {
+import breakpointsFontSize from '../styled/AddSupplierData/breakpointsFontSize';
+import breakpointsMargin from '../styled/AddSupplierData/breakpointsMargin';
 
 const AddSupplierData = () => {
```
In the future this component should be rewritten for inputs that @dominikmlynarczyk prepared.

kacpergadula replied:
It will be corrected

mfurmanit commented on src/components/molecules/PlanRoom/Tittle/index.tsx:
```diff
@@ -0,0 +1,9 @@
+import { Box } from '@chakra-ui/react';
+
+const PlanRoomTitle = () => (
```
The folder and component names should be the same. In addition, you've got a typo there.

kacpergadula replied:
It will be corrected

mfurmanit commented on src/components/molecules/PlanRoom/RoomTables/TableComponentsType/index.tsx:
```diff
@@ -0,0 +1,26 @@
+};
+
+export type TablesType = {
+  [key: string]: { top: number; left: number; text: string; width: number; height: number; borderRadius: number };
```
Properties on the right should be extracted to separate type.
I also posted comments about many common properties in those types that you can extract also.

kacpergadula replied:
It will be separate . Yes, at the moment I have no idea how to do it.

mfurmanit commented on src/components/molecules/PlanRoom/Board/index.tsx:
```diff
@@ -0,0 +1,18 @@
+
+const Board = () => (
+  <Box position="relative" id="board" h="400px" w="100%" border="1px solid black">
+    <Image src={logo} alt="Segun Adebayo" zIndex="0" position="absolute" h="398px" w="100%" />
```
This `alt` property should probably be empty. ;)

mfurmanit replied:
Using `absolute` positioning and `height` for board is not a good idea. 
It should be responsive as other components.

dominikmlynarczyk replied:
First of all, this background should not be a graphic, that is, a `<Image>` component. You should simply use the `background-image` property.

Referring to @mfurmanit reply - I would leave `height` as it is, of course it would be better if it would be responsive (for example `50vh`, but if you do it that way, you will need to probably adjust how tables are rendered (not with fixed values for `top` and `left`, but with percentages). If you can refactor it to be responsive, then great, but if you don't have time, for me it is ok. Setting `w="100%"` is unnecessary.

Please don't add code that will have to be changed later anyway - it's more work, and it would be enough to write it well from the beginning. I'm talking about `border: 1px solid black;` - I know that you used it probably for testing purposes, but you need to change it. See how in other places "Box" components are prepared e.g. basing them on `box-shadow`, which looks much better.

Also, add as little code as possible, so it's easy to reason about. An example would be the use of `id` - this is obviously not a bug, but what is the purpose of adding `id`? This creates a subtle overhead when analyzing the code.

kacpergadula replied:
I did this because the background would not come in correctly on the Box. I knew there was a good path, nevertheless it was not working. I found a way to just replace it. As for the alternative text, it will be corrected.ID referred to another issue that I have already removed from the code. ID will also be removed.

mfurmanit commented on src/components/molecules/PlanRoom/Board/index.tsx:
```diff
@@ -0,0 +1,18 @@
+import { DndProvider } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import Items from 'components/molecules/PlanRoom/RoomTables/Table';
+import logo from 'components/molecules/PlanRoom/IMG/backGroundBoard.png';
```
That should not be called `logo`.

dominikmlynarczyk replied:
We don't have static resources in the project yet. If you add them to the project, we don't do it at the component level, because that won't make the project architecture easier (imagine looking for graphics and having graphics scattered all over the project, at the level of nested directories of small components... not fun 😔).

My suggestion:
Create path `src/assets/images/` and place that background image there. Also, I suggest `kebab-case` for resource names.

kacpergadula replied:
This is the path I will use for the photos -> src/assets/images/

mfurmanit commented on src/components/molecules/PlanRoom/Battons/index.tsx:
```diff
@@ -0,0 +1,43 @@
+import { Button, ButtonGroup, Box } from '@chakra-ui/react';
```
You have a typo in folder name (`Battons` instead of `Buttons`).

dominikmlynarczyk replied:
What is the point of separating this into a separate component anyway? I will point out two main issues:

1. The main factor in separating a given JSX into a separate component is **reusability**. It's not a problem to have more JSX code in one component, as long as it makes sense (it is coherent) and doesn't require breaking it up into reusable (used in many places) smaller components.

2. Why are you styling the Chakra `<Button />` component in a completely different way than other places in the interface (see components in the Menu)? Consider Chakra `<Button />` as atom component and style it accordingly to usages in other places and use Chakra API props wisely. Setting the `width`, `height` is not the best idea, not to mention setting an inline `background-color` for the button - this is very had to maintain such code.

Please remove that component completely, and place Buttons inside parent component.

kacpergadula replied:
As for the buttons, I will leave them in a separate file already. I took the colour from our documentation. The name will be changed

mfurmanit commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+  );
+
+  save = () => {
+    console.log(tables);
```
That should be replaced with API call to Firebase.

dominikmlynarczyk replied:
@mfurmanit I guess it was not part of this ticket.

kacpergadula replied:
@dominikmlynarczyk is right

mfurmanit reviewed (changes requested):
I've added a few more comments in addition to the previous ones. 💪 
Please familiarise with those issues and fix them if you can.

dominikmlynarczyk commented on src/components/molecules/PlanRoom/Battons/index.tsx:
```diff
@@ -0,0 +1,43 @@
+import breakpointsFontSize from 'components/molecules/PlanRoom/styled/breakpointsFontSize';
+import { newTable, saveTables } from 'components/molecules/PlanRoom/RoomTables/Table';
+
+const saveBoard = () => (
```
FYI: Always use _PascalCase_ when naming components.

kacpergadula replied:
Sure, I will improve

kacpergadula replied:
oki

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+import SaveBoard from 'components/molecules/PlanRoom/SaveRoomTables/SaveBoard';
+import { theme } from 'others/theme';
+
+let changeValueTable: Function;
```
What is the purpose of pulling these variables outside of the component? Especially if you are using useCallback then it definitely doesn't seem to make sense.

kacpergadula replied:
I did this because I could not export e.g. addNewTable in order to use it in the place where it was exported at the click of a button

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+let save: Function;
+
+const Items = () => {
+  const [tables, setTables] = useState<TablesType>({
```
Why isn't this an array of objects? Note that you are actually creating something like an associative array anyway, because the keys in your object are numbers that correspond to indexes.

kacpergadula replied:
I will leave it as I have done to move on

dominikmlynarczyk commented on package.json:
```diff
@@ -21,7 +22,10 @@
     "emotion-theming": "^11.0.0",
     "firebase": "^8.4.1",
     "framer-motion": "^3.10.4",
+    "immutability-helper": "^3.1.1",
```
I have analyzed the usage of this package and it boils down to updating a nested object using the `update` function (`<Items />` component). We're not working on _state_ which has a ton of nested objects, so any performance gains are unnoticeable, and introducing a new library is never free.

Besides, the whole issue stems from the wrong structure (nested objects in an object, instead of objects in an array). You need to correct data structure and using `map` would be fine. Then you can remove that library.

kacpergadula replied:
I will leave it as I have done to move on

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+  const moveTable = useCallback(
+    (id: string, left: number, top: number) => {
+      setTables(
+        update(tables, {
```
After changing to array of objects you can replace this with `map` function. I suggest adding `id` or `key` property for each table and then verifying inside `map` if current iteration corresponds to table that is being moved.

kacpergadula replied:
I will leave it as I have done to move on

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+  changeValueTable = useCallback(
+    (id: string, text: string) => {
+      setTables(
+        update(tables, {
```
Same as above.

kacpergadula replied:
I will leave it as I have done to move on

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+  changeValueTableBorderRadius = useCallback(
+    (id: string, borderRadius: number) => {
+      setTables(
+        update(tables, {
```
Same as above.

kacpergadula replied:
I will leave it as I have done to move on

dominikmlynarczyk commented on src/views/Settings/Tables.tsx:
```diff
@@ -1,5 +1,23 @@
-
-const Tables = () => <Heading>MOCK: Stoliki</Heading>;
+import { VStack, Box, Flex } from '@chakra-ui/react';
+import RoomTitle from '../../components/molecules/PlanRoom/Tittle';
```
Use absolute paths, as @mfurmanit mentioned.

kacpergadula replied:
I will use

dominikmlynarczyk commented on src/components/molecules/PlanRoom/Tittle/index.tsx:
```diff
@@ -0,0 +1,9 @@
+import { Box } from '@chakra-ui/react';
+
+const PlanRoomTitle = () => (
+  <Box bg="white" h="4em" w="50%" px="2em" py="0.2em" color="black" fontWeight="bold">
```
Don't use box that way. Also, putting the section title of the admin in a `<div>` is not a best idea.
BTW: Do you need to separate that in another component?

kacpergadula replied:
sure, where I can I will use another component

dominikmlynarczyk commented on src/components/molecules/PlanRoom/styled/breakpointsButton.tsx:
```diff
@@ -0,0 +1,12 @@
+import { createBreakpoints } from '@chakra-ui/theme-tools';
+
+const breakpoints = createBreakpoints({
```
???

kacpergadula replied:
responsiveness to the button

dominikmlynarczyk commented on src/components/molecules/PlanRoom/styled/breakpointsFontSize.tsx:
```diff
@@ -0,0 +1,12 @@
+import { createBreakpoints } from '@chakra-ui/theme-tools';
+
+const breakpointsFontSize = createBreakpoints({
```
???

kacpergadula replied:
responsiveness to the button

dominikmlynarczyk commented on src/others/theme.ts:
```diff
@@ -19,4 +19,9 @@ export const theme = extendTheme({
   },
+  colors: {
+    table: {
+      100: '#211F1F',
```
???
Do you really need to specify that color separately? I guess we can stick with ChakraUI colors.

kacpergadula replied:
as we said, I took it from the documentation

dominikmlynarczyk commented on src/views/Settings/Tables.tsx:
```diff
@@ -1,5 +1,23 @@
+import SaveBoard from '../../components/molecules/PlanRoom/Battons';
+import Board from '../../components/molecules/PlanRoom/Board';
 
+const Tables = () => (
```
I challenge you to simplify that component...  This component suffers from a terrible disease called "DIVitis" 😄.

Also, please use all of the available space within the grid, rather than centering the component in the middle.

kacpergadula replied:
Since there is no limiter on the right side of the board, I cannot use the full random setting when creating a component

dominikmlynarczyk commented on src/components/molecules/Supplier/AddSupplier/index.tsx:
```diff
@@ -1,6 +1,6 @@
-import AddSupplierTitle from 'components/Supplier/AddSupplierTitle';
-import AddSupplierData from 'components/Supplier/AppSupplierData/AddSupplierData';
+import { VStack, Box, Flex } from '@chakra-ui/react';
+import AddSupplierTitle from '../AddSupplierTitle/index';
```
Previous version (before changes) was correct.

kacpergadula replied:
Returns to previous version

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/TableComponentsType/index.tsx:
```diff
@@ -0,0 +1,26 @@
+import { ReactNode } from 'react';
+
+export type TypeItem = {
```
`TypeItem` seems like not the best naming. Also, @mfurmanit can't we simplify types creating intersection with `TableType` (if `TypeItem` is necessary of course).

kacpergadula replied:
these types will be more integrated

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+  );
+};
+
+const tableRadius = (radius: number) => {
```
This function creates code smell. You use magic numbers such as "40". Wouldn't it be easier, if you change the code and instead create function with name `getRadius` or more verbose `shouldProvideRadius`? You would only pass boolean value, and the responsibility of the function is to decide whether to add radius or not.

Example:
`const getRadius = (hasRadius: boolean) => hasRadius ? 40 : 0;`

kacpergadula replied:
Well, yes :D I will use your idea to improve the notation of the function

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/TableComponentsType/index.tsx:
```diff
@@ -0,0 +1,26 @@
+import { ReactNode } from 'react';
+
```
We don't place typings at the component level, do it instead on `models` directory level.

kacpergadula replied:
without adding this it will give me an error

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/TableDNDTypes/index.tsx:
```diff
@@ -0,0 +1,5 @@
+const ItemTypes = {
```
If that type is necessary, place it inside `models` directory.

kacpergadula replied:
I will move this to models

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+
+  addNewTable = (id: number) => {
+    tables[id] = { top: 0, left: 0, text: id.toString(), width: 100, height: 50, borderRadius: 0 };
+    let top = Math.floor(Math.random() * 100) + 40;
```
I guess the reason for this was to not stack new tables, but because you are passing previous index, you move randomly previously added table.

kacpergadula replied:
The issue with the index will be corrected

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+          left = 0;
+        }
+        moveTable(item.id, left, top);
+        return undefined;
```
`return undefined` is probably unnecessary, right? :)

kacpergadula replied:
Right

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+let numberTable: number = 1;
+
+export const newTable = () => {
+  addNewTable(numberTable);
```
This solution is definetely not the best, as you are creating arbitrary counter... If you can't refactor this now, remove that function completely, and just use `addNewTable` and invoke it like this:
`addNewTable(++numberTable)`

kacpergadula replied:
![image](https://user-images.githubusercontent.com/43721352/120938853-4ae9a980-c715-11eb-8a6f-16b4699f55dc.png)

unfortunately ESLint won't let me have it

kacpergadula replied:
Ultimately, I want this data to be stored in a database. it will be known what the value of the counter is

dominikmlynarczyk commented on src/components/molecules/PlanRoom/RoomTables/Table/index.tsx:
```diff
@@ -0,0 +1,182 @@
+let addNewTable: Function;
+let save: Function;
+
+const Items = () => {
```
Maybe rename that component to something like `TableList`?

kacpergadula replied:
I agree with you. I will change it to your proposed

dominikmlynarczyk reviewed (changes requested):
Thanks for your work.
A summary of very important points that you should definitely apply to future changes and work on future tickets:
1. Don't put everything in the `molecules` directory, because that loses the whole point of working with Atomic Design. The goal is that you divide components meaningfully within the project and according to their "size" (which is a big simplification) in the interface place them in the appropriate directory. Placing binary files or types definitions in the molecules directory is a misunderstanding of the atomic design approach.
2. Remember that the Chakra component you use translates ultimately into a DOM tree element and actual HTML markup. Using <Box> everywhere is not a good idea. Most Chakra components extend the Box component, so in most cases you are able to use a dedicated component such as `<Text />`, `<Heading />` or by way of exception, you can use `<Box>` along with the `as` props.
3. Let's stick to the rule - one file, one component. For example, in this PR you have put `<Items />` and `<Table />` components in one file.
4. We use Chakra and we want to make the most of its default appearance of atomic components such as `<Button />` and make reasonable use of styling props. There is virtually never a need to specify the `height` of a component, and you do that a lot. Additionally, you often write `w: 100%` - block elements contain 100% of the parent content space by default, so it just doesn't make sense. We also rely on Chakra colors, because we haven't implemented our own color system - writing an inline color (not stored in the global `theme`) is a mistake, because if you add such a color in many places in the code, you have to "jump" around the whole project to change it later. If you don't feel like a "CSS master", take a look at how other components in the project are written, to make a coherent UI all across our app.

Regarding drag n' drop functionality:
Currently updating data in state at table level only works for `left`, `top` and `borderRadius`, for `text`, `width` and `height` it is not working. Also you randomise position of previous table instead of the newest one.

kacpergadula committed:
refactor: :art: changes after the review ( more to come leter )

kacpergadula committed:
refactor: :art: changes after the review

implementation of suggestions after review

dominikmlynarczyk committed:
Merge branch 'development' into feature/MP-50
